### PR TITLE
Add get received texts function

### PR DESCRIPTION
**Pull request by gov-cjwaszczuk**
- Added getReceivedTexts method so you can get a batch of received texts to your inbound. Takes optional older_than argument that allows them to get 250 from the Id of the message specified.
- Updated documentation
- Added a contributing to match the other clients
- Bumped version

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+export NOTIFY_API_URL="https://example.notify-api.url"
+export API_KEY="example_API_test_key"
+export FUNCTIONAL_TEST_NUMBER="valid mobile number"
+FUNCTIONAL_TEST_EMAIL "valid email address"
+EMAIL_TEMPLATE_ID "valid email_template_id"
+SMS_TEMPLATE_ID "valid sms_template_id"
+LETTER_TEMPLATE_ID "valid letter_template_id"
+EMAIL_REPLY_TO_ID "valid email reply to id"
+SMS_SENDER_ID "valid sms_sender_id - to test sending to a receiving number, so needs to be a valid number"
+API_SENDING_KEY "API_whitelist_key for sending an SMS to a receiving number"
```
Comment (kentsanggds) [inline CONTRIBUTING.md]: should probably `a SMS to`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+INBOUND_SMS_QUERY_KEY "API_test_key to get received text messages - leave blank for local development as cannot test locally"
+```
+
+
+To run the integration tests:
+
+`Make integration-test`
+
+The integration tests are used to test the contract of the response to all the api calls, ensuring the latest version of notifications-api do not break the contract of the notifications-python-client.
```
Comment (kentsanggds) [inline CONTRIBUTING.md]: should be `notifications-node-client`

File: README.md [hunk 3]
```diff
@@ -880,6 +880,73 @@ Otherwise the parameter can be omitted or `undefined` can be passed in its place
 
 </details>
 
+## Get received text messages with pagination
+
+This will return one page of text messages (250) per call. Use the get_received_texts_iterator to retrieve all received texts unpaginated.
```
Comment (kentsanggds) [inline README.md]: This client doesn't have the `get_received_texts_iterator`

File: README.md [hunk 4]
```diff
@@ -880,6 +880,73 @@ Otherwise the parameter can be omitted or `undefined` can be passed in its place
...
+
+
+#### Arguments
+
+<details>
+<summary>
+Click here to expand for more information.
+</summary>
+
+##### `olderThan`
```
Comment (kentsanggds) [inline README.md]: request parameter is `older_than`

File: README.md [hunk 5]
```diff
@@ -880,6 +880,73 @@ Otherwise the parameter can be omitted or `undefined` can be passed in its place
...
+
+If omitted first 250 received text messages are returned. Otherwise the next 250 received text messages older than the given id are returned.
```
Comment (kentsanggds) [inline README.md]: Python client docs were misleading, should be something like `250 of the latest received text messages`

File: spec/integration/schemas/v2/definitions.json [hunk 6]
```diff
@@ -46,5 +46,26 @@
         "subject": {"type": "string"}
     },
     "required": ["body", "subject"]
-  }  
+  },
+  "received_text": {
```
Comment (kentsanggds) [inline spec/integration/schemas/v2/definitions.json]: no longer need `received_text` json block

File: spec/notification.js [hunk 7]
```diff
@@ -375,6 +375,19 @@ describe('notification api', function() {
...
+        .get('/v2/received-text-messages')
+        .reply(200, {"foo":"bar"});
+
+      notifyClient.getReceivedTexts()
+        .then(function(response){
+          expect(response.statusCode).to.equal(200);
+          done();
+        });
+    });
+
```
Comment (kentsanggds) [inline spec/notification.js]: should probably also add an `older_than` unit test for received texts similar to line 260 in this file

File: spec/integration/test.js [hunk 8]
```diff
@@ -281,6 +287,16 @@ describer('notification api with a live service', () => {
       });
     });
 
+    it('get all received texts', (done) => {
```
Comment (leohemsted) [inline spec/integration/test.js]: this `done` pattern sucks if the promise doesn't resolve. If you don't pass in a `done` param, then mocha expects that your test file returns a promise (which it already does), which it just plays out. 

ie: remove `done` from here and the `done()` function call from your `then` callback

File: spec/notification.js [hunk 9]
```diff
@@ -375,6 +375,34 @@ describe('notification api', function() {
...
+    it('should get latest 250 received texts', function(done) {
+
+      notifyAuthNock
+        .get('/v2/received-text-messages')
+        .reply(200, {"foo":"bar"});
+
+      notifyClient.getReceivedTexts()
+        .then(function(response){
+          expect(response.statusCode).to.equal(200);
+          done();
```
Comment (leohemsted) [inline spec/notification.js]: again don't want `done` here. You'll need to `return notifyClient.getReceivedTexts() ...` tho

File: spec/notification.js [hunk 10]
```diff
@@ -375,6 +375,34 @@ describe('notification api', function() {
...
+        var olderThanId = '35836a9e-5a97-4d99-8309-0c5a2c3dbc72';
+
+        notifyAuthNock
+        .get('/v2/received-text-messages?older_than=' + olderThanId)
+        .reply(200, {"foo":"bar"});
+
+        notifyClient.getReceivedTexts(olderThanId)
+        .then(function(response){
+          expect(response.statusCode).to.equal(200);
+          done();
```
Comment (leohemsted) [inline spec/notification.js]: same done here comment

File: spec/integration/schemas/v2/GET_received_text_response.json [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+        "id": {"$ref": "definitions.json#/uuid"},
+        "notify_number": {"type": "string"},
+        "content": {"type": "string"}
+    },
+    "required": [
+        "id", "user_number", "created_at", "service_id",
+        "notify_number", "content"
+    ],
+    "additionalProperties": false
+}
```
Comment (leohemsted) [inline spec/integration/schemas/v2/GET_received_text_response.json]: i don't need you to change these files, but it's probably a good thing to check your sublime settings and ensure that you've got trailing new lines on save turned on. your global sublime config should probably have both of 

```
"ensure_newline_at_eof_on_save": true
"trim_trailing_white_space_on_save": true
```

see https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

**Pull request merged**